### PR TITLE
Add taped test

### PR DESCRIPTION
**Pull request by nebraszka**
See https://github.com/RobotecAI/RGL-blobs/tree/test/integration-using-tape

### CODE REVIEW

File: CMakeLists.txt [hunk 1]
```diff
@@ -21,6 +21,8 @@ set(RGL_AUTO_TAPE_PATH "" CACHE STRING  # STRING prevents from expanding relativ
 # Test configuration
 set(RGL_BUILD_TESTS ON CACHE BOOL
     "Enables building test. GTest will be automatically downloaded")
+set(RGL_BUILD_TAPED_TEST OFF CACHE BOOL
+    "Enables building taped test.")
```
[the remark this answers is not in the document]
Comment (msz-rai) [reply CMakeLists.txt:25]: Also, I'd add info here that a taped test requires closed-source resources to be executed

File: test/src/TapedTest.cpp [hunk 2]
```diff
@@ -0,0 +1,58 @@
...
+#endif
+
+class TapedTest : public RGLTest
+{
+protected:
+	const std::string benchmarkDataDirEnvVariableName = "RGL_TAPED_TEST_DATA_DIR";
+	const float epsilon = 1e-2f;
+};
+
+TEST_F(TapedTest, compare_pcd_files)
```
Comment (msz-rai) [inline test/src/TapedTest.cpp:20]:
```suggestion
TEST_F(TapedTest, awsim-mesh2pcd)
```
I would specify what the use-case the test validates.

File: test/src/TapedTest.cpp [hunk 3]
```diff
@@ -0,0 +1,58 @@
...
+		    "Skipping Taped Test - benchmark data directory must be provided in environment variable '{}'",
+		    benchmarkDataDirEnvVariableName);
+		GTEST_SKIP() << msg;
+	}
+
+	const std::string benchmarkDataDir = std::getenv(benchmarkDataDirEnvVariableName.c_str());
+	const std::string testTapePath{(std::filesystem::path(benchmarkDataDir) / "awsim-mesh2pcd").string()};
+	const std::string expectedOutputPath{
+	    (std::filesystem::path(benchmarkDataDir) / "expected-output" / "awsim-mesh2pcd.pcd").string()};
+	const std::string outputPath{(std::filesystem::current_path() / "output.pcd").string()};
```
Comment (msz-rai) [inline test/src/TapedTest.cpp:36]: I'd avoid hardcoding filenames (e.g. `awsim-mesh2pcd`). Constexpr variables at the top of the test would improve readability and simplify test modifications if names would change.

**Review by msz-rai — COMMENTED** [review]
Nice integration :+1: 
I have left the comments to address.

File: setup.py [hunk 4]
```diff
@@ -261,6 +280,53 @@ def install_ros2_deps(cfg):
...
+def clone_taped_test_data_repo(cfg):
+    run_subprocess_command(
+        f"git clone -b {cfg.TAPED_TEST_DATA_BRANCH} --single-branch --depth 1 {cfg.TAPED_TEST_DATA_REPO} {cfg.TAPED_TEST_DATA_DIR}")
+    os.chdir(cfg.TAPED_TEST_DATA_DIR)
+    # Set up git-lfs for this repository
+    run_subprocess_command("git-lfs install && git-lfs pull")
+
+
+def is_taped_data_up_to_date():
+    result = subprocess.Popen("git fetch --dry-run --verbose", shell=True, stdout=subprocess.PIPE,
```
[the remark this answers is not in the document]
Comment (nebraszka) [reply setup.py:300]: `git-lfs fetch` is not necessary. Changes to LFS-tracked files alter their pointer files, and git fetch detects these updates.

File: CMakeLists.txt [hunk 5]
```diff
@@ -21,6 +21,8 @@ set(RGL_AUTO_TAPE_PATH "" CACHE STRING  # STRING prevents from expanding relativ
 # Test configuration
 set(RGL_BUILD_TESTS ON CACHE BOOL
     "Enables building test. GTest will be automatically downloaded")
+set(RGL_BUILD_TAPED_TESTS OFF CACHE BOOL
+    "Enables building taped test.")
```
Comment (msz-rai) [inline CMakeLists.txt:25]: GTest should also be downloaded for `RGL_BUILD_TAPED_TESTS`. Please take a look at [external/CMakeLists.txt](https://github.com/RobotecAI/RobotecGPULidar/blob/develop/external/CMakeLists.txt#L53)

File: setup.py [hunk 6]
```diff
@@ -233,15 +252,15 @@ def install_pcl_deps(cfg):
         f"{os.path.join(cfg.VCPKG_DIR, cfg.VCPKG_EXEC)} install --clean-after-build pcl[core,visualization]:{cfg.VCPKG_TRIPLET}")
 
 
-def has_colcon():
-    process = subprocess.Popen("colcon --help", shell=True, stderr=subprocess.DEVNULL, stdout=subprocess.DEVNULL)
+def is_command_available(command):
+    process = subprocess.Popen(f"which {command}", shell=True, stderr=subprocess.DEVNULL, stdout=subprocess.DEVNULL)
```
Comment (msz-rai) [inline setup.py:256]: AFAIK, there is no `which` program on Windows. I'd stick with checking the pure command here and adding `--help` option to `colcon` and `git-lfs` commands below.

File: test/taped_test/CMakeLists.txt [hunk 7]
```diff
@@ -0,0 +1,29 @@
...
+
+set(RGL_TAPED_TEST_FILES
+    src/AwsimMeshToPcdTest.cpp
+)
+
+include(GoogleTest)
+
+add_executable(RobotecGPULidar_taped_test ${RGL_TAPED_TEST_FILES})
+
+find_package(GTest REQUIRED)
```
Comment (msz-rai) [inline test/taped_test/CMakeLists.txt:15]: GTest should already be downloaded and propagated via fetch content.

**Review by msz-rai — APPROVED** [review]
LGTM :+1:

**Pull request merged**